config/samples/bpfman.io_v1alpha1_bpfapplication.yaml

```diff
@@ -9,50 +9,113 @@ spec:
   nodeselector: {}
   bytecode:
     image:
-      url: quay.io/bpfman-bytecode/go-app-counter:latest
+      url: quay.io/bpfman-bytecode/app-test:latest
+      # imagepullpolicy: Always
   programs:
-    - type: Kprobe
+    - bpffunctionname: kprobe_test
+      type: Kprobe
       kprobe:
-        bpffunctionname: kprobe_counter
-        func_name: try_to_wake_up
-        offset: 0
-        retprobe: false
-    - type: Tracepoint
+        attach_points:
+          - func_name: try_to_wake_up
+            offset: 0
+            retprobe: false
+    - bpffunctionname: kretprobe_test
+      type: Kprobe
```

Comment:
Do we need to expose `type` here?
It can be inferred based on which of the optional fields are set so it seems redundant

Reply:
This is implemented this as union, and Type is the "unionDiscriminator".  This tells us which of the "unionMembers" will be populated, and validation logic ensures that if the Type is X, the program info for X is populated.  @msherif1234 used this pattern for the original BpfApplication CRD. There may be another way to implement the API, but this works.